test/ex_doc/autolink_test.exs

```diff
@@ -161,11 +161,55 @@ defmodule ExDoc.AutolinkTest do
       assert autolink(~m"[custom text](`:lists.all/2`)") ==
                ~m"[custom text](http://www.erlang.org/doc/man/lists.html#all-2)"
 
-      # TODO: with custom links and backticks there are no false positives (you
-      #       always mean to link) so we should always warn on mismatches?
-      #       Though backticks are markdown specific, is that ok?
-      # assert_warn(fn ->
-      assert_unchanged(~m"[custom text](`Unknown`)")
+      captured =
+        assert_warn(fn ->
+          assert_unchanged(~m"[custom text](`Unknown`)")
+        end)
+
+      assert captured =~
+               "documentation references module or file \"`Unknown`\" but it doesn't exist"
```

Comment:
> "documentation references module or file (...)

I'd remove the "or file" piece, since custom links are special I believe we can be more strict:

1. `` [foo](`Foo`) `` - if something is within backticks, it is a module/function/type/etc, and if we cannot find it, we warn
2. `` [foo](Foo) `` - if something is without backticks, then it can only be an extra, and if we don't have it, we warn

IMHO the warning is not going to be as useful but still useful enough, and we'd keep things a bit simpler. WDYT?

thank you for handling this TODO, btw!

Reply:
I agree with simplifying things. My only concern is that people will get confused when using custom links swap the terms (I did that myself A LOT).

Reply:
done